### PR TITLE
Additional tests of classification performance and weight properties

### PR DESCRIPTION
**Pull request by SkBlaz**
- parameterized training data generator to simplify running many `ad hoc` experiments if required
- added a bash script that trains fw on generated data, stores the outputs, computes relevant classification metrics (on training data, intentionally) and alerts the user if something is off --- including properties of prediction files, as well as classification capabilities vs. random must be in good shape/of quality. Current experiments indicate that random vs. `fw` balanced accuracy on training data ((sensitivity + specificity) / 2) difference is more than 0.2, hence this is the current margin considered for the test to pass.
- added an action that runs this for each commit, so we know where things went south if that's the case
 
The plan is to include this as a git action so that for each new binary we have a few learning-level sanity checks conducted. Currently, we included the main properties we are most interested in with each new version; more can be added should the need arise.

### CODE REVIEW

**Review by adischw — APPROVED** [review]

**Comment by bbenshalom** [comment]
Great work @SkBlaz ! 
What was the motivation to use a bash script for the tests rather than rust?
Plus, from what I've seen, the term `benchmarking` in rust is used for performance (as in latency, cpu time, etc.), and here the test is for the accuracy (for lack of a better term) of the predictions. Consider making that distinction in this PR

**Review by bbenshalom — APPROVED** [review]

**Comment by SkBlaz** [comment]
Thanks @bbenshalom, appreciate you had a look! Yes, so:

- The decision for bash vs. e.g., sklearn-based script (more supported metrics out-of-the-box) vs. native rust -> having a self-contained bash file seemed like the most "native" solution to the issue of being able to test the properties of predictions more or less anywhere (e.g., across multiple machines)

- At the same time, it's rather straight-forward to add/remove tests and metrics should the need arise. Further, running this on different data sets with different fw configs at this point requires minimal, straight-forward modification (not necessarily the case with native rust that one would have to recompile on a given machine which immediately implies e.g., `rustup` is set up fine there etc.)

"Benchmark" can indeed mean different things in the context of this repo, made the distinction a bit clearer as suggested (the flow action is now called learning test, albeit it in fact conducts a mini _learning_ benchmark - not a performance one indeed).
